[javac] Enum valueOf causing `Cannot read the array length because "methods" is null` NPE

The following code causes an NPE in vscode-java, while computing inlay hints:
```java
package org.acme;

public class MyClass {

    private enum Color {
        black, white, blue,red ,green;
    }

    public MyClass(String foo){
        Color.valueOf(foo); //This line causes the NPE
    }
}
```

```
[Error - 10:35:21 AM] Request textDocument/inlayHint failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot read the array length because "methods" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
Caused by: java.lang.NullPointerException: Cannot read the array length because "methods" is null
	at org.eclipse.jdt.internal.javac.dom.JavacMethodBinding.getJavaElement(JavacMethodBinding.java:171)
	at org.eclipse.jdt.ls.core.internal.handlers.InlayHintVisitor.getParameterNames(InlayHintVisitor.java:212)
	at org.eclipse.jdt.ls.core.internal.handlers.InlayHintVisitor.resolveInlayHints(InlayHintVisitor.java:165)
	at org.eclipse.jdt.ls.core.internal.handlers.InlayHintVisitor.visit(InlayHintVisitor.java:87)
	at org.eclipse.jdt.core.dom.MethodInvocation.accept0(MethodInvocation.java:220)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3343)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChild(ASTNode.java:3391)
	at org.eclipse.jdt.core.dom.ExpressionStatement.accept0(ExpressionStatement.java:136)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3343)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:3414)
	at org.eclipse.jdt.core.dom.Block.accept0(Block.java:128)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3343)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChild(ASTNode.java:3391)
	at org.eclipse.jdt.core.dom.MethodDeclaration.accept0(MethodDeclaration.java:677)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3343)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:3414)
	at org.eclipse.jdt.core.dom.TypeDeclaration.accept0(TypeDeclaration.java:498)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3343)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:3414)
	at org.eclipse.jdt.core.dom.CompilationUnit.accept0(CompilationUnit.java:258)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3343)
	at org.eclipse.jdt.ls.core.internal.handlers.InlayHintsHandler.inlayHint(InlayHintsHandler.java:81)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$56(JDTLanguageServer.java:1184)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
```

--- a/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacMethodBinding.java
+++ b/org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacMethodBinding.java
@@ -168,7 +168,7 @@ public abstract class JavacMethodBinding implements IMethodBinding {
 									Signature.createTypeSignature(resolveTypeName(t, true), true))
 							.toArray(String[]::new);
 					IMethod[] methods = currentType.findMethods(currentType.getMethod(getName(), parametersResolved));
-					if (methods.length > 0) {
+					if (methods != null && methods.length > 0) {
 						return methods[0];
 					}
 					var parametersNotResolved = this.methodSymbol.params().stream()
@@ -178,7 +178,7 @@ public abstract class JavacMethodBinding implements IMethodBinding {
 									Signature.createTypeSignature(resolveTypeName(t, false), false))
 							.toArray(String[]::new);
 					methods = currentType.findMethods(currentType.getMethod(getName(), parametersNotResolved));
-					if (methods.length > 0) {
+					if (methods != null && methods.length > 0) {
 						return methods[0];
 					}
 				}

